waveorder/io/upti.py

```diff
@@ -0,0 +1,191 @@
+import numpy as np
+import os
+import re
+import zarr
+import tifffile as tiff
+import glob
+from waveorder.io.reader_interface import ReaderInterface
+
+
+class UPTIReader(ReaderInterface):
+
+    """
+    Reader for UPTI raw data.  Accepts both new live UPTI and older UPTI format.
+    """
+
+    def __init__(self, folder: str, extract_data: bool = False):
+
+        # check if folder exists
+        if not os.path.isdir(folder):
+            raise ValueError("folder does not exist")
+
+        # Initializate data parameters
+        self.data_folder = folder
+        self.files = glob.glob(os.path.join(folder, '*.tif'))
```

Comment:
Some of my super old files are saved under ".tiff" rather than ".tif". An example is under `/gpfs/CompMicro/rawdata/hummingbird/LiHao/20200223_LCD_63x_147_oil_test/Mouse_brain_702_ROI2/data_2/`. It will throw error because of this mismatch. Potentially an `if` statement may do it okay.